pkg/stack/unwind/executable.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2022 The Parca Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package unwind
+
+import (
+	"debug/elf"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/go-version"
+	"github.com/xyproto/ainur"
+)
+
+func HasFramePointers(executable string) (bool, error) {
+	elf, err := elf.Open(executable)
+	if err != nil {
+		return false, fmt.Errorf("failed to open ELF file for path %s: %w", executable, err)
+	}
+	defer elf.Close()
+
+	compiler := ainur.Compiler(elf)
+	// Go 1.7 [0] enabled FP for x86_64. arm64 got them enabled in 1.12 [1].
+	//
+	// Note: we don't take into account applications that use cgo yet.
+	// If the non Go bits aren't compiled with frame pointers, too,
+	// unwinding will fail. In the future might add the unwind information
+	// for these bits of executable code.
+	//
```

Comment:
That's a good point, but right now we are only focused on the executable passed to see whether it was compiled frame pointers or not. While it's true that statically compiled Go binaries with CGO will return true here but need special handling, this is something for the callsite to handle, and where it can log the appropriate error message

Reply:
Yup, sounds good to log where the function will be called.